src/window.ts

```diff
@@ -522,25 +524,61 @@ export class ShellWindow {
     private window_raised() {
         this.restack(RESTACK_STATE.RAISED);
         this.show_border();
+        if (this.ext.conf.move_pointer_on_switch && !this.pointer_already_on_window()) {
```

Comment:
Is not moving the pointer if the pointer's already somewhere on the activated window what we really want? On one hand, it prevents unnecessary cursor movement if the user just wants the cursor near the window, but on the other hand, it makes the behavior a little less consistent/predictable.

Video to show what I mean: https://www.dropbox.com/s/rzoeicovtm33ps0/shell-pr646-screencast1.mp4?dl=0 Notice how the cursor is moved to the bottom right of the Files app the first time I switch to it, but not the second time, because the cursor happened to already be on top of the Files app the second time.

Another example of the same inconsistency when switching workspaces (without using Alt-Tab): https://www.dropbox.com/s/eyb5o5zmx9en0rn/shell-pr646-screencast2.mp4?dl=0 

If we're good with this as intended behavior (which is disabled by default anyway), and if "switch" is the correct terminology to use for this option (as opposed to "activation" or "focus"), then this is fine to merge to dev.

Reply:
I'm open to discussions on the behavior. The current behavior always places the pointer on the top left to aid mouse movements on ultrawide displays. The top left corner of an application usually has the most clickable controls, and is generally out of the way

Reply:
@jacobgkau Yes in fact it's the intended behavior. The reason why I enabled this is because if you switch application just with your mouse by cliking on the window, if I keep behavior to move cursor it becomes very weird because your pointer suddenly moves to another position, which is not expected. As a user of this feature for a week now I have to say I'm pretty satisfy with this behavior and don't feel inconsistency but yes here is my humble opinion. If you have any other suggestions I'm also open to discussion :)

Reply:
> if you switch application just with your mouse by cliking on the window, if I keep behavior to move cursor it becomes very weird because your pointer suddenly moves to another position

I agree, we wouldn't want that, so the current behavior's fine. Thanks for the explanation.